packages/contractkit/src/wallets/signers/rpc-signer.ts

```diff
@@ -103,11 +105,20 @@ export class RpcSigner implements Signer {
   getNativeKey = () => this.account
 
   async unlock(passphrase: string, duration: number) {
-    const unlocked = await this.callAndCheckResponse(RpcSignerEndpoint.UnlockAccount, [
-      this.account,
-      passphrase,
-      duration,
-    ])
+    let unlocked = false
+    try {
+      unlocked = await this.callAndCheckResponse(RpcSignerEndpoint.UnlockAccount, [
+        this.account,
+        passphrase,
+        duration,
+      ])
+    } catch (error) {
+      // Only throw errors other than expected unlock error
+      if (!error?.message?.toLowerCase().includes(UNLOCK_ACCOUNT_EXPECTED_ERROR)) {
```

Comment:
why is this error expected? and should we not set the `unlockTime` below if it's not successful?

Reply:
+1
can this try catch be raised above this interface?

Reply:
It's because the response from the RPC call includes an error, so in `callAndCheckResponse` below an error is thrown. That method is used in several places so I figured here was a reasonable place to intercept the error. Open to other suggestions

Reply:
I think you need an extra  `?.` so includes isnt called on null

Reply:
Connor's original point still stands that if we suppress this error, we need to not set the `unlockTime` and `unlockDuration`
My main qualm is for other consumers of this interface (of which there are none right now 🙄) this behavior could be confusing

Reply:
Thanks for the eyes on this guys! I'll fix the `unlockDuration` and error safe accessing now.

RE interface, I'm open to changing this back but I'd argue this is more intuitive for consumers. Before, this could never return false. Only true or it throws. So why return anything?

Reply:
And the problem with it never returning false is that consumers have not easy way to differentiate a wrong password from an error, other than catching and inspecting the error for this message `could not decrypt key with given password`.